sdk/core/Azure.Core.Experimental/src/MessageWithMetadata.cs

```diff
@@ -18,7 +20,19 @@ public class MessageWithMetadata
         /// <summary>
         /// Gets or sets the message content type.
         /// </summary>
-        public virtual string? ContentType { get; set; }
+        public virtual ContentType? ContentType
+        {
+            get => ContentTypeCore;
+            set => ContentTypeCore = value;
+        }
+
+        /// <summary>
```

Comment:
This is pretty comprehensive.   If we carry this into derived types, I think this addresses my concern about having something local that explains the structure.